pkg/cluster/rook_ceph_test.go

```diff
@@ -264,3 +270,1056 @@ func TestController_GetRookVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestController_UseNodesForStorage(t *testing.T) {
+	type args struct {
+		rookVersion semver.Version
+		names       []string
+	}
+	tests := []struct {
+		name                 string
+		nodes                []string
+		rookResources        []runtime.Object
+		args                 args
+		want                 int
+		wantStorageScopeSpec cephv1.StorageScopeSpec
+		wantErr              bool
+	}{
+		{
+			name:  "storage nodes increase from useAllNodes to 1, rook version 1.9.12",
```

Comment:
Could we add the expected result in the name such as should pass when .... ?

Reply:
Done